packages/annotation/src/core/toolOperation/cuboidOperation.ts

```diff
@@ -522,6 +613,73 @@ class CuboidOperation extends BasicToolOperation {
     const { selectedCuboid } = this;
     if (selectedCuboid) {
       this.renderSingleCuboid(selectedCuboid);
+      this.renderToggleButton();
+    } else {
+      this.toggleButtonInstance?.clearCuboidButtonDOM();
+      this.toggleButtonInstance = undefined;
+    }
+  }
+
+  // Executed when the mouse is moved in and out to toggle the glossy button
+  public updateMouseOperation(type: 'in' | 'out') {
+    if (type === 'in') {
+      // Show crosshairs
+      this.setForbidCursorLine(true);
+      // Restricted mouse action
+      this.setForbidOperation(true);
+      this.setShowDefaultCursor(true);
+    } else {
+      this.setForbidCursorLine(false);
+      this.setShowDefaultCursor(false);
+      this.setForbidOperation(false);
+    }
+  }
+
+  // Switching glossy surfaces
+  public toggleGlossySide(direction: ECuboidDirection) {
+    if (this.cuboidList && this.selectedCuboid) {
+      this.cuboidList = this.cuboidList.map((i) => (i.id === this.selectedCuboid?.id ? { ...i, direction } : i));
+      this.render();
+    }
+  }
+
+  // Get the points of the corresponding faces by direction
+  public getPointListsByDirection(direction: ECuboidDirection, frontPoints: IPlanePoints, backPoints: IPlanePoints) {
```

Comment:
You can use an object if your params is more than 2.

Reply:
And this function needs to move to `src/utils/tool/CuboidUtils`. Because it is a pure function

Reply:
And you can just transfer a param (cuboid: ICuboid)